test/service/test_manifest_async.py

```diff
@@ -183,10 +180,12 @@ def test(self, mock_uuid, mock_helper):
                     }
                     path = '/manifest/files'
                     object_key = 'some_object_key'
-                    manifest_url = furl(url=self.base_url,
-                                        path=path,
-                                        args=dict(params, objectKey=object_key)).url
-                    url = self.base_url + ('/fetch' + path if fetch else path)
+                    manifest_url = self.base_url.set(path=path,
+                                                     args=dict(params, objectKey=object_key)).url
+                    url = self.base_url.set(path=path)
+                    if fetch:
+                        url.path.segments.insert(0, 'fetch')
+                    url = url.url
 
                     for i, expected_status in enumerate(3 * [301] + [302]):
                         response = requests.get(url, params=params, allow_redirects=False)
```

Comment:
Didn't you mean to set the query parameters via the URL?

Reply:
I did, I must have missed this one. Some usages were a bit harder to discern.